### ISSUE TITLE
Local variable `isa` referenced before assignment

### ISSUE THREAD

**Issue by WeiN76LQh** [droe/binja-blocks#14]
```
[ScriptingProvider] stack block 181a1de14: UnboundLocalError: local variable 'isa' referenced before assignment
[ScriptingProvider] Traceback (most recent call last):
[ScriptingProvider]   File "/Users/0xalsr/Library/Application Support/Binary Ninja/plugins/binja-blocks/blocks.py", line 745, in annotate_stack_block_literal
[ScriptingProvider]     bl = BlockLiteral.from_stack(bv, block_literal_insn, block_literal_var, sym_addrs)
[ScriptingProvider]   File "/Users/0xalsr/Library/Application Support/Binary Ninja/plugins/binja-blocks/blocks.py", line 344, in from_stack
[ScriptingProvider]     return cls(bv, is_stack_block, bl_insn, isa, flags, reserved, invoke, descriptor)
[ScriptingProvider] UnboundLocalError: local variable 'isa' referenced before assignment
```

This was found running the `Annotate all blocks` command on DSC for iOS 18.1.1 22B91 iPhone17,2 with the `libsystem_c.dylib` (loaded by default), `Foundation` and `libsystem_blocks.dylib` images loaded.

This error has occurred in a number of other instances.

**Comment by droe** [comment]
Can repro w/macOS 14.x DSC.

**Comment by droe** [comment]
Root cause is that reloading the stack variable insn when there are multiple insns with the same address may fail to pick up the correct insn:

```
181d365ec            int64_t var_f0_1 = 0
181d36608            int64_t var_158_1 = 0
181d36608            struct Block_literal var_150
181d36608            var_150.isa = &__NSConcreteStackBlock
181d3660c            var_150.flags = 0xc2000000
181d3660c            var_150.reserved = 0
181d36620            var_150.invoke = ____NSLexiconIterateFormsOfWord_block_invoke.8
181d36620            var_150.descriptor = &___block_descriptor_56_e8_32o40o48r_e15_"NSString"8?0l
```

The reloaded instruction ends up as `int64_t var_158_1 = 0` instead of `struct Block_literal var_150`, resulting in the code to look for the wrong variable identifier, for which it will not find any isa field assignment.

**Comment by droe** [comment]
Fixed by 9b496f8.
